include/ParameterSet.hpp

```diff
@@ -308,8 +308,8 @@ class ParameterSet {
     ParameterSet& addInt8(const std::vector<int8_t>& vals) throw (voltdb::ParamMismatchException) {
         validateType(WIRE_TYPE_TINYINT, true);
         m_buffer.ensureRemaining(6 + static_cast<int32_t>(sizeof(int8_t) * vals.size()));
-        m_buffer.putInt8(WIRE_TYPE_ARRAY);
-        m_buffer.putInt8(WIRE_TYPE_TINYINT);
+        // Convert array of TINYINT to VARBINARY.
+        m_buffer.putInt8(WIRE_TYPE_VARBINARY);
```

Comment:
The server converts array of tinyint to varbinary.  If we want the messages from the client to the server to be bytewise equal we have to do this conversion.  If we want to compare messages by parsing the messages and looking for this we could do that, but it would require an extra day of messing around to create such a function.